bin/shutter

```diff
@@ -4624,14 +4633,22 @@ sub STARTUP {
 
 		#settings
 		eval {
-			my ($tmpfh, $tmpfilename) = tempfile(UNLINK => 1);
-			XMLout(\%settings, OutputFile => $tmpfilename);
+			# only write settings to file, if they are sane
+			if ($settings_sane) {
+				my ($tmpfh, $tmpfilename) = tempfile(UNLINK => 1);
+				XMLout(\%settings, OutputFile => $tmpfilename);
 
-			#and finally move the file
-			mv($tmpfilename, $settingsfile);
+				#and finally move the file
+				mv($tmpfilename, $settingsfile);
+			} else {
+				die "Please correct your settings!\n\n";
+			}
 		};
 		if ($@) {
-			$sd->dlg_error_message($@, $d->get("Settings could not be saved!"));
+			if ($settings_sane) {
+				# eval died due to file writing error, not bad settings, show an error message
```

Comment:
Do "if" outside eval, then you won't need to do such checks

Reply:
I'm not quite sure what you mean, this "if" is already outside of eval, isn't it? If you mean the other "if" which is inside of eval, it is there to only write to file when the settings are sane. The file writing was in an eval originally and I just left it this way.

Reply:
Never mind, I removed the whole if checks, as you recommended in your second comment!